include/ddc/kernels/splines/spline_evaluator_2d.hpp

```diff
@@ -1289,15 +1378,54 @@ class SplineEvaluator2D
         ddc::Coordinate<continuous_dimension_type1> const coord_eval_interest1(coord_eval);
         ddc::Coordinate<continuous_dimension_type2> const coord_eval_interest2(coord_eval);
 
-        if constexpr (std::is_same_v<EvalType1, eval_type>) {
+        if constexpr (!in_tags_v<deriv_dim1, deriv_dims>) {
             jmin1 = ddc::discrete_space<bsplines_type1>().eval_basis(vals1, coord_eval_interest1);
-        } else if constexpr (std::is_same_v<EvalType1, eval_deriv_type>) {
-            jmin1 = ddc::discrete_space<bsplines_type1>().eval_deriv(vals1, coord_eval_interest1);
+        } else {
+            if (order1 == 0) {
+                jmin1 = ddc::discrete_space<bsplines_type1>()
+                                .eval_basis(vals1, coord_eval_interest1);
+            } else if (order1 == 1) {
+                jmin1 = ddc::discrete_space<bsplines_type1>()
+                                .eval_deriv(vals1, coord_eval_interest1);
+            } else {
+                std::array<double, 3 * (bsplines_type1::degree() + 1)> derivs1_ptr;
+                Kokkos::mdspan<
+                        double,
+                        Kokkos::extents<std::size_t, bsplines_type1::degree() + 1, 3>> const
+                        derivs1(derivs1_ptr.data());
+
+                jmin1 = ddc::discrete_space<bsplines_type1>()
+                                .eval_basis_and_n_derivs(derivs1, coord_eval_interest1, order1);
+
+                for (std::size_t i = 0; i < bsplines_type1::degree() + 1; ++i) {
+                    vals1[i] = DDC_MDSPAN_ACCESS_OP(derivs1, i, order1);
+                }
+            }
```

Comment:
I would be in favor of only keeping the cases `order1 > 0 && order1 < 3`. The case `order1 == 0`, can be represented by not adding the dimension in the discrete element. I wonder if the case `order1 == 1` is necessary, I guess we can consider it an optimization path ?

@EmilyBourne Any opinion ?